Possible uninitialized variables - compiler warnings (release build, CentOS 7)

**Is your feature request related to a problem? Please describe.**
Various warnings on CentOS 7 when BUILDTYPE=release:
```
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function ‘CFE_SB_TransmitMsg’:
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1433:16: error: ‘RouteId’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         Status = CFE_SB_TransmitBufferFull(BufDscPtr, RouteId, MsgId);
                ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function ‘CFE_SB_SendMsg’:
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1433:16: error: ‘RouteId’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1395:25: note: ‘RouteId’ was declared here
     CFE_SBR_RouteId_t   RouteId;
                         ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function ‘CFE_SB_PassMsg’:
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1433:16: error: ‘RouteId’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
         Status = CFE_SB_TransmitBufferFull(BufDscPtr, RouteId, MsgId);
                ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1395:25: note: ‘RouteId’ was declared here
     CFE_SBR_RouteId_t   RouteId;
                         ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function ‘CFE_SB_RcvMsg’:
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:2045:43: error: ‘RcvStatus’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
                 CFE_EVS_SendEventWithAppID(CFE_SB_Q_RD_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB_Global.AppId,
                                           ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:1823:28: note: ‘RcvStatus’ was declared here
     int32                  RcvStatus;
                            ^
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c: In function ‘CFE_SB_ReceiveBuffer’:
/home/jhageman/cFS/cFS-GitHub/cfe/fsw/cfe-core/src/sb/cfe_sb_api.c:2045:43: error: ‘RcvStatus’ may be used uninitialized in this function [-Werror=maybe-uninitialized]
                 CFE_EVS_SendEventWithAppID(CFE_SB_Q_RD_ERR_EID,CFE_EVS_EventType_ERROR,CFE_SB_Global.AppId,
                                           ^
```

**Describe the solution you'd like**
Initialize variables where needed

**Describe alternatives you've considered**
None

**Additional context**
Compiler warnings, vs static analysis warnings reported in #1185

**Requester Info**
Jacob Hageman - NASA/GSFC

--- a/fsw/cfe-core/src/inc/private/cfe_sbr.h
+++ b/fsw/cfe-core/src/inc/private/cfe_sbr.h
@@ -38,6 +38,13 @@
 #include "cfe_msg_typedefs.h"
 #include "cfe_platform_cfg.h"
 
+/*
+ * Macro Definitions
+ */
+
+/** \brief Invalid route id */
+#define CFE_SBR_INVALID_ROUTE_ID ((CFE_SBR_RouteId_t) {.RouteId = 0})
+
 /******************************************************************************
  * Type Definitions
  */

--- a/fsw/cfe-core/src/sb/cfe_sb_api.c
+++ b/fsw/cfe-core/src/sb/cfe_sb_api.c
@@ -1397,6 +1397,7 @@ int32  CFE_SB_TransmitMsg(CFE_MSG_Message_t *MsgPtr, bool IncrementSequenceCount
 
     PendingEventID = 0;
     BufDscPtr = NULL;
+    RouteId = CFE_SBR_INVALID_ROUTE_ID;
 
     Status = CFE_SB_TransmitMsgValidate(MsgPtr, &MsgId, &Size, &RouteId);
 
@@ -1886,6 +1887,7 @@ int32  CFE_SB_ReceiveBuffer(CFE_SB_Buffer_t **BufPtr,
     BufDscPtr = NULL;
     DestPtr = NULL;
     BufDscSize = 0;
+    RcvStatus = OS_SUCCESS;
     
     /*
      * Check input args and see if any are bad, which require

--- a/fsw/cfe-core/src/sb/cfe_sb_priv.c
+++ b/fsw/cfe-core/src/sb/cfe_sb_priv.c
@@ -536,15 +536,7 @@ void CFE_SB_RemoveDestNode(CFE_SBR_RouteId_t RouteId, CFE_SB_DestinationD_t *Nod
 int32 CFE_SB_ZeroCopyReleaseAppId(CFE_ES_AppId_t AppId)
 {
     CFE_SB_BufferLink_t *NextLink;
-
-    /* Using a union type permits promotion of CFE_SB_BufferLink_t* to CFE_SB_BufferD_t*
-     * without violating aliasing rules or alignment rules.  Note that the first element
-     * of CFE_SB_BufferD_t is a CFE_SB_BufferLink_t, which makes this possible. */
-    union LocalBufDesc
-    {
-        CFE_SB_BufferLink_t As_Link;
-        CFE_SB_BufferD_t    As_Dsc;
-    } *BufPtr;
+    CFE_SB_BufferD_t    *DscPtr;
 
     /*
      * First go through the "ZeroCopy" tracking list and find all nodes
@@ -560,16 +552,18 @@ int32 CFE_SB_ZeroCopyReleaseAppId(CFE_ES_AppId_t AppId)
         while(!CFE_SB_TrackingListIsEnd(&CFE_SB_Global.ZeroCopyList, NextLink))
         {
             /* Get buffer descriptor pointer */
-            BufPtr = (union LocalBufDesc*)NextLink;
+            /* NOTE: casting via void* here rather than CFE_SB_BufferD_t* avoids a false
+             * alignment warning on platforms with strict alignment requirements */
+            DscPtr = (void *)NextLink;
 
             /* Read the next link now in case this node gets moved */
             NextLink = CFE_SB_TrackingListGetNext(NextLink);
 
             /* Check if it is a zero-copy buffer owned by this app */
-            if (CFE_RESOURCEID_TEST_EQUAL(BufPtr->As_Dsc.AppId, AppId))
+            if (CFE_RESOURCEID_TEST_EQUAL(DscPtr->AppId, AppId))
             {
                 /* If so, decrement the use count as the app has now gone away */
-                CFE_SB_DecrBufUseCnt(&BufPtr->As_Dsc);
+                CFE_SB_DecrBufUseCnt(DscPtr);
             }
         }
 

--- a/modules/sbr/private_inc/cfe_sbr_priv.h
+++ b/modules/sbr/private_inc/cfe_sbr_priv.h
@@ -31,12 +31,6 @@
  */
 #include "private/cfe_sbr.h"
 
-/*
- * Macro Definitions
- */
-
-/** \brief Invalid route id */
-#define CFE_SBR_INVALID_ROUTE_ID ((CFE_SBR_RouteId_t) {.RouteId = 0})
 
 /******************************************************************************
  * Function prototypes

